chore(docs): Update tutorial part eight to use an SEO component (#17730)
gatsbyjs/gatsby: docs/tutorial/part-eight/index.md

npm install --save gatsby-plugin-react-helmet react-helmet
```

> 2.  Make sure you have a `description` and an `author` configured inside your `siteMetadata` object. Also, you have to add the `gatsby-plugin-react-helmet` plugin to the `plugins` array in your `gatsby-config.js` file.

Reviewer: ```suggestion
2.  Make sure you have a `description` and an `author` configured inside your `siteMetadata` object. Also, add the `gatsby-plugin-react-helmet` plugin to the `plugins` array in your `gatsby-config.js` file.
```

      meta={[
        {
          name: `description`,
          content: metaDescription,
        },
        {
          property: `og:title`,
          content: title,
        },
        {
          property: `og:description`,
          content: metaDescription,
        },
        {
          property: `og:type`,
          content: `website`,
        },
        {
          name: `twitter:card`,
          content: `summary`,
        },
        {
          name: `twitter:creator`,
          content: site.siteMetadata.author,
        },
        {
          name: `twitter:title`,
          content: title,
        },
        {
          name: `twitter:description`,
          content: metaDescription,
        },
      ].concat(meta)}
    />
  )
}

SEO.defaultProps = {
  lang: `en`,
  meta: [],
  description: ``,
}

SEO.propTypes = {
  description: PropTypes.string,
  lang: PropTypes.string,
  meta: PropTypes.arrayOf(PropTypes.object),
  title: PropTypes.string.isRequired,
}

export default SEO
```

The above code sets up defaults for your most common metadata tags and provides you an `<SEO>` component to work with in the rest of your project. Pretty cool, right?

> 4.  Now, you can use the `<SEO>` component in your templates and pages and pass props to it. For example, add it to you `blog-posts.js` file like so:

Reviewer 2: ```suggestion
4.  Now, you can use the `<SEO>` component in your templates and pages and pass props to it. For example, add it to your `blog-posts.js` file like so:
```

```jsx:title=blog-posts.js
import React from "react"
import { graphql } from "gatsby"
import Layout from "../components/layout"
// highlight-next-line
import SEO from "../components/seo"

export default ({ data }) => {
  const post = data.markdownRemark
  return (
    <Layout>

>       // highlight-next-line

Reviewer 2: ```suggestion
      // highlight-start
```

      <SEO
        title={post.frontmatter.title}
        description={post.frontmatter.excerpt}

>       />

Reviewer 2: ```suggestion
      />
      // highlight-end
```

    `gatsby-plugin-offline`,
    // highlight-next-line
    `gatsby-plugin-react-helmet`,
  ],
}
```

3. In the `src/components` directory, create a file called `seo.js` and add the following:

> ```jsx:title=seo.js

Reviewer: ```suggestion
```jsx:title=src/components/seo.js
```

      meta={[
        {
          name: `description`,
          content: metaDescription,
        },
        {
          property: `og:title`,
          content: title,
        },
        {
          property: `og:description`,
          content: metaDescription,
        },
        {
          property: `og:type`,
          content: `website`,
        },
        {
          name: `twitter:card`,
          content: `summary`,
        },
        {
          name: `twitter:creator`,
          content: site.siteMetadata.author,
        },
        {
          name: `twitter:title`,
          content: title,
        },
        {
          name: `twitter:description`,
          content: metaDescription,
        },
      ].concat(meta)}
    />
  )
}

SEO.defaultProps = {
  lang: `en`,
  meta: [],
  description: ``,
}

SEO.propTypes = {
  description: PropTypes.string,
  lang: PropTypes.string,
  meta: PropTypes.arrayOf(PropTypes.object),
  title: PropTypes.string.isRequired,
}

export default SEO
```

The above code sets up defaults for your most common metadata tags and provides you an `<SEO>` component to work with in the rest of your project. Pretty cool, right?

> 4.  Now, you can use the `<SEO>` component in your templates and pages and pass props to it. For example, add it to your `blog-posts.js` file like so:

Reviewer: ```suggestion
4.  Now, you can use the `<SEO>` component in your templates and pages and pass props to it. For example, add it to your `blog-post.js` template like so:
```

> ```jsx:title=blog-posts.js

Reviewer: The rest of the tutorial uses a different structure:
```suggestion
```jsx:title=src/templates/blog-post.js
```

import React from "react"
import { graphql } from "gatsby"
import Layout from "../components/layout"
// highlight-next-line
import SEO from "../components/seo"

export default ({ data }) => {
  const post = data.markdownRemark
  return (
    <Layout>
      // highlight-start
      <SEO
        title={post.frontmatter.title}
        description={post.frontmatter.excerpt}
      />
      // highlight-end
      <div>
        <h1>{post.frontmatter.title}</h1>
        <div dangerouslySetInnerHTML={{ __html: post.html }} />
      </div>
    </Layout>
  )
}

export const query = graphql`
  query($slug: String!) {
    markdownRemark(fields: { slug: { eq: $slug } }) {
      html
      frontmatter {
        title
        // highlight-next-line
        excerpt
      }
    }
  }
`
```

> The above example is based off the [Gatsby Starter Blog](/starters/gatsbyjs/gatsby-starter-blog/). You can see that by passing props to the `<SEO>` component, you can dynamically change the metadata for a post. In this case, the blog post `title` and `excerpt` (if it exists in the blog post markdown file) will be pulled in instead of the `title` and `description` properties of the `siteMetadata` object in your `gatsby-config.js` file.

Reviewer 3: I think this is a bit easier to read.

```suggestion
The above example is based off the [Gatsby Starter Blog](/starters/gatsbyjs/gatsby-starter-blog/). By passing props to the `<SEO>` component, you can dynamically change the metadata for a post. In this case, the blog post `title` and `excerpt` (if it exists in the blog post markdown file) will be used instead of the default `siteMetadata` properties in your `gatsby-config.js` file.
```
